pkg/db/interface.go

```diff
@@ -90,6 +93,26 @@ func getDbName() string {
 	return fmt.Sprintf(dbNameTmpl, dbPass)
 }
 
+func openSQLConn(driverName string, dataSourceName string, interval time.Duration,
+	timeout time.Duration) (*sql.DB, error) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	startTime := time.Now()
+	for {
+		select {
+		case <-ticker.C:
+			if db, err := sql.Open(driverName, dataSourceName); err == nil {
+				if err = db.Ping(); err == nil {
+					return db, nil
+				}
+			}
+		case <-time.After(timeout - time.Since(startTime)):
```

Comment:
I think a new timer instance is created on every connection failure and they are leaked. What do you think?

Reply:
If connection attempt fails consecutively, I think just one instance lives until return. There seems no chance of entering this loop in a short period